### DISCUSSION TITLE
Trying to use AMD GPU in Docker container on Unraid

### DISCUSSION THREAD

**Discussion by Olivier-Roy** [blakeblackshear/frigate#14046] [Hardware Acceleration Support]
Hi everyone, I am trying to configure Frigate with hardware acceleration using an old GPU lying around but I can’t get it working. It shows up in Unraid System Devices as:
 
[1002:68d9] 01:00.0 VGA compatible controller: Advanced Micro Devices, Inc. [AMD/ATI] Redwood PRO [Radeon HD 5550/5570/5630/6510/6610/7570]
 
It should be compatible with Frigate according to the documentation:
 
https://docs.frigate.video/configuration/hardware_acceleration/#amdati-gpus-radeon-hd-2000-and-newer-gpus-via-libva-mesa-driver
 
Here is my Docker container configuration:
 
![image](https://github.com/user-attachments/assets/d6c66bf7-fa20-43ab-af12-1219399107d9)
 
Here is my config.yml file:
 
![image](https://github.com/user-attachments/assets/febe8460-0ab8-4186-a83f-e442b0f6f6d7)
 
Here are the logs when I start the container:
 
![image](https://github.com/user-attachments/assets/f3b39e99-ce68-430f-a7e1-373ed668e547)

**Comment by NickM-27** [comment]
run `ls /dev/dri` in the unraid cli and show what returns. Are you sure you enabled the driver by installing radeon-top?

**Reply by Olivier-Roy** [reply]
![image](https://github.com/user-attachments/assets/43ba88a8-9bc2-4a3a-b34b-600fe13d8a37)

Yes I installed radeontop:
![image](https://github.com/user-attachments/assets/7cc144a2-bd12-40bc-9aca-68608096048e)

**Reply by NickM-27** [reply]
I'd suggest using the `Hardware Info` button in the system page to see what hardware is detected

**Reply by Olivier-Roy** [reply]
![Capture d’écran 2024-09-30 111715](https://github.com/user-attachments/assets/07c96c5f-223e-473a-8620-32984ad5f377)

**Reply by NickM-27** [reply]
that is not what I requested, I am talking about the Frigate system page to be clear

**Reply by Olivier-Roy** [reply]
Oh sorry, here's what I get when I click on "Hardware Info" button:
![Capture d’écran 2024-09-30 112145](https://github.com/user-attachments/assets/d854b3dd-2ea1-40e5-9e0c-679f0dc3beb2)
![Capture d’écran 2024-09-30 112133](https://github.com/user-attachments/assets/da3584bc-5246-4dd5-a38d-2df88ffc9d5f)

**Reply by NickM-27** [reply]
is it scrollable at all? there should be more to it, you can also just copy and paste it

**Reply by Olivier-Roy** [reply]
I copied the message: (It's not scrollable)

{"return_code":1,"stderr":"error:XDG_RUNTIME_DIRisinvalidornotsetintheenvironment.nerror:can'tconnecttoXserver!nerror:failedtoinitializedisplay","stdout":""}

**Reply by NickM-27** [reply]
seems it is not seeing the GPU, please follow these steps so we can double check:

1. click on the frigate icon in docker page and then press `console`
2. run `vainfo`
3. run `ls /dev/dri`
4. paste a screenshot of the whole thing

**Reply by Olivier-Roy** [reply]
![image](https://github.com/user-attachments/assets/b23ad4e3-7c3f-43d4-9034-c850d64a6fc5)

**Reply by Olivier-Roy** [reply]
{"return_code":0,"stderr":"","stdout":"vainfo:VA-APIversion:1.17(libva2.10.0)nvainfo:Driverversion:MesaGalliumdriver22.3.6forAMDREDWOOD(DRM2.50.0/6.1.49-Unraid,LLVM15.0.6)nvainfo:SupportedprofileandentrypointsnVAProfileMPEG2Simple:tVAEntrypointVLDnVAProfileMPEG2Main:tVAEntrypointVLDnVAProfileVC1Simple:tVAEntrypointVLDnVAProfileVC1Main:tVAEntrypointVLDnVAProfileVC1Advanced:tVAEntrypointVLDnVAProfileH264ConstrainedBaseline:tVAEntrypointVLDnVAProfileH264Main:tVAEntrypointVLDnVAProfileH264High:tVAEntrypointVLDnVAProfileNone:tVAEntrypointVideoProc"}

![image](https://github.com/user-attachments/assets/aae03afd-543d-49ce-9081-e2c800319e09)

![image](https://github.com/user-attachments/assets/d56280d8-73ec-4b51-af5f-cbd36d11dae7)

![image](https://github.com/user-attachments/assets/f480e500-2d93-4c1a-9ece-71bcaac2d671)

Wait now it seems like it detects my GPU, but I still get these two warnings.
Is it because I'm missing a "detectors" section in my [config.yml](https://github.com/blakeblackshear/frigate/discussions/14046#discussion-7249126) file?

**Reply by NickM-27** [reply]
1. you should set 

```
ffmpeg:
  hwaccel_args: preset-vaapi
```

at the global part of the config not just on the camera level

and for the second warning, you still need something to run object detection besides the CPU

**Reply by Olivier-Roy** [reply]
I thought I could use my GPU for object detection?
If I can't, I guess I can use OpenVINO with my Intel CPU?
What should I add in my config.yml to make it work?

**Reply by NickM-27** [reply]
OpenVino supports intel GPUs as well as Intel CPUs officially, AMD GPUs are only supported for object detection (using ROCm) in the upcoming 0.15 version

**Reply by Olivier-Roy** [reply]
Ok so I will try to use my Intel CPU for object detection for now. I changed my config.yml file, but I still get errors I don't understand:

![image](https://github.com/user-attachments/assets/1665006f-c400-445e-a300-1ad3f7fc1194)

![image](https://github.com/user-attachments/assets/ab93ccdd-413a-4e14-89ed-4c08dd0a4f1b)

**Reply by NickM-27** [reply]
I have not seen that error before so I am not sure

**Reply by Olivier-Roy** [reply]
I found this, and my container seems to run the same openvino version (2024.1.0-15008-f4afc983258-releases/2024/1)

https://github.com/openvinotoolkit/openvino/issues/24333

**Reply by NickM-27** [reply]
Seems being a warning it is a nonissue
